src/library/Button/index.d.ts

```diff
@@ -1 +1,36 @@
-declare module '@targetx/mineral-ui/Button';
+import {
+  ElementType,
+  ReactElement,
+  ReactNode,
+  Ref,
+  SyntheticEvent
+} from 'react';
+
+export declare namespace Button {
+  type Size = 'small' | 'medium' | 'large' | 'jumbo';
+  type Variant = 'danger' | 'grayscale' | 'success' | 'warning';
+
+  export interface Props {
+    as?: ElementType;
+    children?: ReactNode;
+    circular?: boolean;
+    css?: { [key: string]: any };
+    disabled?: boolean;
+    forwardedRef?: Ref<any>;
+    fullWidth?: boolean;
+    href?: string;
+    iconEnd?: ReactElement;
+    iconStart?: ReactElement;
+    justifyContent?: string;
+    minimal?: boolean;
+    onClick?: (event: SyntheticEvent) => void;
+    primary?: boolean;
+    size?: Size;
+    type?: string;
+    variant?: Variant;
+    width?: string;
+  }
+}
+
+export declare function Button(props: Button.Props): ReactElement;
```

Comment:
This doesn't match what @tx-trent did for `Icon`.
```
export declare const Icon: (props: Icon.Props) => React.ReactElement<any, string | ((props: any) => React.ReactElement<any, string | any | (new (props: any) => React.Component<any, any, any>)> | null) | (new (props: any) => React.Component<any, any, any>)>;
```

Not sure if we need all of that, but let's be consistent and use the same approach.